### PR TITLE
added domain checking utility

### PR DESCRIPTION
**Pull request by y4utj4**
@wolfthefallen @zeroSteiner 
**Description**
Checks to see if the sending email's domain is valid or not. This is done by first trying to resolve the domain's IP address, then if it can, checking and displaying some whois information through the python-whois library.

**Dependencies**
- [x] python-whois

**Testing Procedures**
- [x] Install plugin
- [x] Install pywhois library (python3 -m pip install python-whois)
- [x] Enable Plugin
- [x] Send test email, an alert will generate with the appropriate information.

### CODE REVIEW

File: client/domain_check.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+try:
+	import whois
+except ImportError:
+	has_pywhois = False
+else:
+	has_pywhois = True
+
+class Plugin(plugins.ClientPlugin):
+    authors = ['Jeremy Schoeneman']  # the plugins author
```
Comment (wolfthefallen) [inline client/domain_check.py]: inconsitant between spaces and tabs for indentation

Comment (wolfthefallen) [reply client/domain_check.py]: Through the whole file.. Please reform to only using tabs

File: client/domain_check.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    version = '1.0'            # (optional) specify this plugin's version
+    # this is the primary plugin entry point which is executed when the plugin is enabled
+    req_packages = {
+		'pywhois': has_pywhois
+	}
+
+    def initialize(self):
+        mailer_tab = self.application.main_tabs['mailer']
+        self.signal_connect('send-precheck', self.signal_check_domain, gobject=mailer_tab)
+
```
Comment (wolfthefallen) [inline client/domain_check.py]: need a `return True` here for the plugin to properly initialize

File: client/domain_check.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        user,at,domain = email.partition('@')
+        try:
+            self.logger.debug('checking email domain')
+            result = getaddrinfo(domain, None)
+            if result:
+                try:
+                    info = whois.whois(domain)
+                    self.logger.info('email domain valid')
+                    gui_utilities.show_dialog_info(
+                    'Email Domain Valid',
```
Comment (wolfthefallen) [inline client/domain_check.py]: lines 42 - 46 need to have an extre indentation so it looks like:
```python
gui_utilities.show_dialog_info(
    'Email Domain Valid',
    self.application.get_active_window(),
    'Domain registered to: ' + info.name + '\n',
    'Name Servers: ' + info.name_servers + '\n',
    'Contact Email: ' + info.emails
)
```

File: client/domain_check.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                    self.application.get_active_window(),
+                    'Domain registered to: ' + info.name + '\n',
+                    'Name Servers: ' + info.name_servers + '\n',
+                    'Contact Email: ' + info.emails
+                    )
+                except Exception as err:
+                    self.logger.info('whois lookup unavailable')
+                    gui_utilities.show_dialog_info(
+                    'Email Domain Valid',
```
Comment (wolfthefallen) [inline client/domain_check.py]: 51 - 53 needs an extra indentation to show it is part of the `gui_utilities.show_dialog_info` like above

**Comment by wolfthefallen** [comment]
Once you address the issues, I'll test again and land.

**Comment by zeroSteiner** [comment]
Any word on this @wolfthefallen?

**Comment by wolfthefallen** [comment]
Commited in 6cb55d001816cddb9df39b81acf5cf84b55454d4 into dev branch

**Pull request closed without merging**